### ISSUE TITLE
Wrong requests to HB server

### ISSUE THREAD

**Issue by malex984** [hilbert/hilbert-heartbeat#23]
The following  works as expected:
```
./hb_preload.sh ./browser.sh -l "file://$PWD/index.html?HB_APP_ID=app&HB_URL=http://127.0.0.1:8888&n=star"
```

i.e. HB ping GET requests are as follows: `/hb_ping?6000&appid=app&cache_buster=1234567890* HTTP/1.1`

BUT is i change `&` to `\&` in the passed argument:
```
./hb_preload.sh ./browser.sh -l "file://$PWD/index.html?HB_APP_ID=app\&HB_URL=http://127.0.0.1:8888\&n=star"
```
then HB server pings will look as follows: `//hb_ping?6000&appid=app\&cache_buster=1234567890* HTTP/1.1`

BUT if i now remove the URL tail-argument as follows:
```
./hb_preload.sh ./browser.sh -l "file://$PWD/index.html?HB_APP_ID=app\&HB_URL=http://127.0.0.1:8888"
```

then everything is back to normal: HB pings are `/hb_ping?6000&appid=app\&cache_buster=1234567890* HTTP/1.1`

Kiosk-browser actually gets and loads exactly the same url strings (with `\&` in it)

**Comment by malex984** [comment]
@porst17 if `hilbert-heartbeat.js` gets confused by parameters given by `\&` and reacts by prefixing everything with extra back-slash then why the 3rd case (with back-slash, without 3rd parameter) works as expected (see the next comment)...?!

**Comment by malex984** [comment]
btw, the problem of HB server are not prefixed parameters but two slashes in HB_PING messages: `//hb_ping?6000&...`

**Comment by porst17** [comment]
According to https://www.ietf.org/rfc/rfc2396.txt (page 10), backslashes (`\`) can't be used in URIs (and therefore URLs). So basically, you are reporting strange behavior for invalid input, which is to be expected.

The behavior of Chrome is to automatically replace `\` by `/`, e.g. to support loading local files on Windows. Just enter `file://\` into the location bar and see what happens (spoiler: it will be turned into `file:///` and lists files in `/` on your system).

So the heartbeat JS library will parse `HB_APP_ID=app\&HB_URL=http://127.0.0.1:8888\&n=star` into
```
HB_APP_ID=app\
HB_URL=http://127.0.0.1:8888\
n=star
```
`HB_URL` will just be used as is when connecting to the heartbeat server, but Chrome then automatically turns
```
http://127.0.0.1:8888\/hb_ping?6000&appid=app\&cache_buster=1234567890*
```
into
```
http://127.0.0.1:8888//hb_ping?6000&appid=app\&cache_buster=1234567890*
```

So the actual problem is not in the heartbeart JS library, but in the invalid parameters that are submitted to it.

**Comment by malex984** [comment]
Ok, but then i do not see why everything still works if there is no last parameter `n=star`..?

**Comment by malex984** [comment]
Ok. i see now

**Comment by porst17** [comment]
Because without the last parameter you have `HB_URL=http://127.0.0.1:8888`, which is totally fine.

**Comment by malex984** [comment]
it is resolved now. 

NOTE: and heartbeat server was improved thanks to this issue.

**Comment by porst17** [comment]
I assume this is the commit that fixes it: d2cb47966148bcb30e0057ec6befb02592aff44f

**Comment by malex984** [comment]
and i also checked that the current wrappers have no issues with `&` in the URL argument
